File: java/src/main/java/com/experian/dvs/client/address/suggestions/format/Suggestion.java
package com.experian.dvs.client.address.suggestions.format;

import com.experian.dvs.client.address.format.Address;
import com.experian.dvs.client.address.format.AddressComponents;
import com.experian.dvs.client.address.format.AddressFormatted;
import com.experian.dvs.client.address.format.AddressMetadata;
import com.experian.dvs.client.server.address.suggestions.RestApiSuggestionsFormatSuggestion;

import java.util.Objects;
import java.util.Optional;

public class Suggestion {
    private final String globalAddressKey;
    private final Address address;
    private final AddressFormatted addressFormatted;
    private final AddressComponents components;
    private final AddressMetadata metadata;

    public Suggestion(final RestApiSuggestionsFormatSuggestion suggestion) {
        if (suggestion != null) {
            this.globalAddressKey = Objects.toString(suggestion.getGlobalAddressKey(), "");
            this.address = suggestion.getAddress() != null ? new Address(suggestion.getAddress()) : null;
            //only one currently supported so just pick the first
            this.addressFormatted = suggestion.getAddressesFormatted() != null  && !suggestion.getAddressesFormatted().isEmpty() ? new AddressFormatted(suggestion.getAddressesFormatted().get(0)) : null;
            this.components = suggestion.getComponents() != null ? new AddressComponents(suggestion.getComponents()) : null;
            this.metadata = suggestion.getMetadata() != null ? new AddressMetadata(suggestion.getMetadata()) : null;
        } else {
            this.globalAddressKey = "";
            this.address = null;
            this.addressFormatted = null;
            this.components = null;
            this.metadata = null;
        }
    }

    public String getGlobalAddressKey() {
        return globalAddressKey;
    }

    public Address getAddress() {
        return address;
    }

    public AddressFormatted getAddressFormatted() {
        return addressFormatted;
    }

    public Optional<AddressMetadata> getMetadata() {
        return Optional.ofNullable(metadata);
    }

    public Optional<AddressComponents> getComponents() {
        return Optional.ofNullable(components);
    }

}

File: csharp/DVSClient/DVSClient/address/suggestions/Suggestion.cs
using DVSClient.Address.Format;
using DVSClient.Server.Address.Suggestions;

namespace DVSClient.Address.Suggestions
{
    public class Suggestion
    {
        public string? GlobalAddressKey { get; }
        public Format.Address? Address { get; }
        public AddressFormatted? AddressFormatted { get; }
        public AddressComponents? Components { get; }
        public AddressMetadata? Metadata { get; }

        public Suggestion(RestApiSuggestionsFormatSuggestion suggestion)
        {
            if (suggestion != null)
            {
                GlobalAddressKey = suggestion.GlobalAddressKey;
                Address = suggestion.Address != null ? new Format.Address(suggestion.Address) : null;
                AddressFormatted = suggestion.AddressesFormatted != null ? new AddressFormatted(suggestion.AddressesFormatted.First()) : null;
                Components = suggestion.Components != null ? new AddressComponents(suggestion.Components) : null;
                Metadata = suggestion.Metadata != null ? new AddressMetadata(suggestion.Metadata) : null;
            }
            else
            {
                GlobalAddressKey = string.Empty;
                Address = null;
                AddressFormatted = null;
                Components = null;
                Metadata = null;
            }
        }
    }
}